app/models/subject.rb

```diff
@@ -1,6 +1,6 @@
 class Subject < ApplicationRecord
   has_many :lessons
-  has_many :teachers
+  has_many :teachers, class_name: "Teacher"
```

Comment:
why do you need this? it is by default

Reply:
i saw one example and tried to write as it was there. but i checked without it and it works. I removed.